### ISSUE TITLE
Horizontal scrolling in text files

### ISSUE THREAD

**Issue by michael-lazar** [skyjake/lagrange#313]
Hi!

I have the setting "wrap plain text: no" turned on in order to view .txt files containing ASCII art and other preformatted text documents. It would be helpful to be able to horizontally scroll text files in this mode, the same way that scrolling works in preformatted text blocks.

In a similar vein, I would also suggest that *not* wrapping plain text files should be the default setting, since they're almost always hard wrapped and will look weird unless you apply a reflowing algorithm.

Here's an example link for reference

gemini://ascii.mozz.us:7070/mirrors/incredibleart/files/scarecrow_sig_files.txt

**Comment by skyjake** [comment]
I agree. However, the current situation is basically a compromise. The scrollable document view in Lagrange has been implemented only for vertical scrolling, which means that long lines would not be viewable at all in the app without breaking the lines.

Enabling horizontal scrolling right now would lead to very inefficient drawing. I'm not particularly happy with the partial solution for scrolling wide preformatted runs, as that introduces quite a bit of complexity in to the system (#44).

I will likely end up rethinking how the document view works so it can scroll both ways without a large performance penalty.
